src/plugins/orangehrmDashboardPlugin/Dao/ChartDao.php

```diff
@@ -85,13 +87,50 @@ private function getEmployeeDistributionQueryBuilderWrapper(EmployeeSearchFilter
 
         if (!is_null($employeeSearchFilterParams->getSubunitId())) {
             $q->andWhere($q->expr()->in('subunit.id', ':subunitId'))
-                ->setParameter('subunitId', $employeeSearchFilterParams->getSubunitIdChain());
+                ->setParameter(
+                    'subunitId',
+                    $employeeSearchFilterParams->getSubunitIdChain()
+                );
         }
 
         if (is_null($employeeSearchFilterParams->getSubunitId())) {
             $q->andWhere($q->expr()->isNull('employee.subDivision'));
         }
 
+        $q->andWhere($q->expr()->isNull('employee.employeeTerminationRecord'));
+        $q->andWhere($q->expr()->isNull('employee.purgedAt'));
+
         return $this->getQueryBuilderWrapper($q);
     }
+
+    /**
+     * @return int
+     */
+    public function getTotalActiveEmployeeCount(): int
+    {
+        $q = $this->createQueryBuilder(Employee::class, 'employee');
+        $q->andWhere($q->expr()->isNull('employee.employeeTerminationRecord'));
+        $q->andWhere($q->expr()->isNull('employee.purgedAt'));
+
+        return $this->count($q);
+    }
+
+    /**
+     * @return LocationEmployeeCount[]
+     */
+    public function getEmployeeDistributionByLocation(): array
+    {
+        $select = 'NEW ' . LocationEmployeeCount::class .
+            '(location.id, location.name, COUNT(employee.empNumber))';
+        $q = $this->createQueryBuilder(EmpLocations::class, 'el')
+            ->leftJoin('el.employee', 'employee')
+            ->leftJoin('el.location', 'location')
+            ->select($select);
+        $q->andWhere($q->expr()->isNull('employee.employeeTerminationRecord'));
+        $q->andWhere($q->expr()->isNull('employee.purgedAt'));
+        $q->addGroupBy('location.id');
+        $q->addOrderBy('COUNT(employee.empNumber)', 'DESC');
```

Comment:
```suggestion
        $q->addOrderBy('empCount', ListSorter::DESCENDING);
```
most probably will fail, but will try this